### PR TITLE
Universal Deeplinking support

### PR DESCRIPTION
**Pull request by kschmidtdev**
Universal Deeplinking support

JIRA: **https://mobify.atlassian.net/browse/MSG-396**
Linked PRs: **https://github.com/mobify/astro/pull/513**

## Changes
- Add iOS code and reference entitlements file to be ready for universal deeplinking (required for app push test enable triggering)

## How to test-drive this PR
- Generate a project, set up a non-wildcard app id and provisioning profile for it, and update config json in https://github.com/mobify/deeplinkr

## TODOS:
~~- [ ] Change works in both Android and iOS.~~

- [x] CHANGELOG.md has been updated.

- [x] Run the generator to make sure it still functions correctly.
- [ ] +1 from an engineer on the Astro team.

~~- [ ] Both tab layout and drawer layout are fully functional in ios. (Set `iosUsingTabLayout` in `baseConfig`)~~

### CODE REVIEW

**Comment by jeremywiebe** [comment]
👍

File: ios/scaffold/scaffold.entitlements [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>application-identifier</key>
+	<string>com.mobify.astro.scaffold</string>
```
Comment (jeremywiebe) [inline ios/scaffold/scaffold.entitlements]: Does the generator correctly update this app ID when generating an app?

Comment (kschmidtdev) [inline ios/scaffold/scaffold.entitlements]: It does 💥

File: ios/scaffold/AppDelegate.swift [hunk 2]
```diff
@@ -76,6 +76,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
     }
 
+    func application(application: UIApplication, continueUserActivity userActivity: NSUserActivity, restorationHandler: ([AnyObject]?) -> Void) -> Bool {
+        if let astroViewController = astroViewController {
```
Comment (jeremywiebe) [inline ios/scaffold/AppDelegate.swift]: Given that we just need to hand off the `userActivity` here. What do you think about making this more concise?
```swift
return astroViewController?.continueUserActivity(userActivity) ?? false 
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ## To be released
 - [Android] Updated to be compatible with Android Cordova v5.1.1
 - [iOS] Include pushclient module in iOS build
+- Add support for iOS universal deeplinking
```
Comment (jeremywiebe) [inline CHANGELOG.md]: Nitpick... What about prefixing the entry with the platform like other entries above it?  
```
- [iOS] Add support for universal links
```

**Comment by jeremywiebe** [comment]
👍   Nice work @kschmidtdev

**Pull request merged**